patches/tModLoader/Terraria/ModLoader/EquipLoader.cs

```diff
@@ -298,5 +305,487 @@ public static class EquipLoader
 				texture?.FrameEffects(player, type);
 			}
 		}
+
+		internal static ModAccessorySlot GetModAccessorySlot(int slot) {
+			ModContent.TryFind<ModAccessorySlot>(ModAccessorySlot.moddedAccSlots[slot], out ModAccessorySlot mAccSlot);
+			if (mAccSlot == null) {
+				mAccSlot = new UnloadedAccessorySlot(slot);
+			}
+
+			return mAccSlot;
+		}
```

Comment:
I think a different class would be better for these methods. The role of `EquipLoader` is just mapping item ids to equip types and ids. Mod accessory slot loading/hooks should go in a new class.